egui/src/widgets/progress_bar.rs

```diff
@@ -0,0 +1,124 @@
+use crate::*;
+
+/// A simple progress bar.
+pub struct ProgressBar {
+    progress: f32,
+    desired_width: Option<f32>,
+    text: String,
+    animate: bool,
+}
+
+impl ProgressBar {
+    pub fn new(progress: f32) -> Self {
+        Self {
+            progress: progress.clamp(0.0, 1.0),
+            desired_width: None,
+            text: "".to_string(),
+            animate: false,
+        }
+    }
+
+    /// The desired width of the bar. Will use all horizonal space if not set.
+    pub fn desired_width(mut self, desired_width: f32) -> Self {
+        self.desired_width = Some(desired_width);
+        self
+    }
+
+    /// A custom text to display on the progress bar.
+    #[allow(clippy::needless_pass_by_value)]
+    pub fn text(mut self, text: impl ToString) -> Self {
+        self.text = text.to_string();
+        self
+    }
+
+    /// Whether to display a loading animation. Note that this require the UI to be redrawn.
+    /// Defaults to `false`.
```

Comment:
It is an interesting question what the default should be here.

It wastes some CPU in reactive mode using `eframe`, but basically anywhere else it is nice with an animation.

In the future we may want to have a global egui option for "do you want animations or not?", but for now... either `false` or `true` work for me to be honest.

Reply:
The animation now plays on hover, I'll leave it up to you to decide whether it should be animated by default.